crates/hyperqueue/src/dashboard/ui/widgets/cpu_util_table.rs

```diff
@@ -0,0 +1,59 @@
+use std::cmp;
+use tui::layout::{Constraint, Rect};
+use tui::widgets::{Cell, Row, Table};
+
+use crate::dashboard::ui::styles;
+use crate::dashboard::ui::terminal::DashboardFrame;
+use crate::dashboard::ui::widgets::progressbar::{
+    get_progress_bar_color, render_progress_bar_at, ProgressPrintStyle,
+};
+
+const CPU_METER_LEN: u8 = 18;
+
+pub fn render_cpu_util_table(
+    usages: &[f32],
+    rect: Rect,
+    frame: &mut DashboardFrame,
+    constraints: &[Constraint],
+) {
+    let rows: Vec<Row> = usages
+        .chunks(constraints.len())
+        .map(|cpu_usage_column_dat| {
+            let columns: Vec<Cell> = cpu_usage_column_dat
+                .iter()
+                .map(|utilization| {
+                    let progress = utilization / 100.00;
+                    Cell::from(render_progress_bar_at(
+                        *utilization / 100.00,
+                        CPU_METER_LEN,
+                        ProgressPrintStyle::default(),
+                    ))
+                    .style(get_progress_bar_color(progress))
+                })
+                .collect();
+            Row::new(columns)
+        })
+        .collect();
+
+    let title = styles::table_title(format!("Worker CPU Utilization({} CPUs)", usages.len()));
+    let body_block = styles::table_block_with_title(title);
+    let table = Table::new(rows)
+        .block(body_block)
+        .highlight_style(styles::style_table_highlight())
+        .style(styles::style_table_row())
+        .widths(constraints);
+
+    frame.render_widget(table, rect);
+}
+
+pub fn get_column_constraints(rect: Rect, num_cpus: usize) -> Vec<Constraint> {
```

Comment:
Could you please add a doc. comment that says what does this function do? Thanks.